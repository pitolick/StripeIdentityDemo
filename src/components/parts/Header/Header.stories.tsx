import React from 'react'
import { Meta, StoryObj } from '@storybook/react'
import { Header } from './Header'

const meta: Meta<typeof Header> = {
  title: 'parts/Header',
  component: Header,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
  decorators: [(storyFn) => <div className="relative h-80 scale-100 overflow-scroll">{storyFn()}</div>],
}

export default meta
type Story = StoryObj<typeof Header>

/**
 * デフォルト
 */
export const Default: Story = {}

/**
 * 子要素あり
 */
export const WithChildren: Story = {
  args: {
    children: <div className="container text-center">children</div>,
  },
}

/**
 * 追従あり
 */
export const Fixed: Story = {
  args: {
    children: <div className="container text-center">children</div>,
    isFixed: true,
  },
}
