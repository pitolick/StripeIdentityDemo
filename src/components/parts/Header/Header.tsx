/**
 * ヘッダー
 * @module Header
 */
import React from 'react'

/**
 * ヘッダーのコンポーネントのプロパティ
 */
interface HeaderProps {
  /**
   * 子要素
   */
  children?: React.ReactNode
  /**
   * 追従するかどうか
   */
  isFixed?: boolean
}

/**
 * ヘッダーのコンポーネント
 * @param {HeaderProps} props -  ヘッダーコンポーネントのプロパティ
 * @returns {React.FC} -  ヘッダーコンポーネント
 */
export const Header: React.FC<HeaderProps> = ({ children, isFixed = false, ...headerProps }) => {
  // 追従するかどうか
  const fixedClass = isFixed ? 'fixed top-0 left-0 w-full z-50' : ''

  return (
    <header id="header" className={`${fixedClass}`} {...headerProps}>
      {children}
    </header>
  )
}
