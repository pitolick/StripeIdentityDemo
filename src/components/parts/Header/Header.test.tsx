import React from 'react'
import { render } from '@testing-library/react'
import { Header } from './Header'

describe('Headerコンポーネント', () => {
  it('子要素が正しく表示されること', () => {
    const { container } = render(
      <Header>
        <div>テスト</div>
      </Header>,
    )
    expect(container).toMatchSnapshot()
  })

  it('isFixedがtrueの場合、fixedクラスが追加されること', () => {
    const { container } = render(<Header isFixed={true} />)
    expect(container).toMatchSnapshot()
  })

  it('isFixedがfalseの場合、fixedクラスが追加されないこと', () => {
    const { container } = render(<Header isFixed={false} />)
    expect(container).toMatchSnapshot()
  })
})
