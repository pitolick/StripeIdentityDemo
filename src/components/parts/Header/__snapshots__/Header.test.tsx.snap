// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headerコンポーネント isFixedがfalseの場合、fixedクラスが追加されないこと 1`] = `
<div>
  <header
    class=""
    id="header"
  />
</div>
`;

exports[`Headerコンポーネント isFixedがtrueの場合、fixedクラスが追加されること 1`] = `
<div>
  <header
    class="fixed top-0 left-0 w-full z-50"
    id="header"
  />
</div>
`;

exports[`Headerコンポーネント 子要素が正しく表示されること 1`] = `
<div>
  <header
    class=""
    id="header"
  >
    <div>
      テスト
    </div>
  </header>
</div>
`;
